### DISCUSSION TITLE
Problem redirecting to another module with react-router-dom

### DISCUSSION THREAD

**Discussion by ahmetartana** [module-federation/module-federation-examples#2822] [Q&A] [answered]
I want to redirect from one app to another app with Navigate but I get an issue like below.
 
No routes matched location "/dashboard/home"   (I already have "/dashboard/home"  but in another mf module)

This module is taken as an example :  https://github.com/module-federation/module-federation-examples/tree/master/react-nested-routers
My code instance is here:  https://github.com/ahmetartana/artana-mf/blob/main/packages/auth/src/pages/Login/Login.tsx

Is it possible use react-router-dom navigate function?

**Comment by yasiinakbulut58** [comment, marked as answer]
Hi @ahmetartana,

you can listen router changes on subApp and invoke a callback method.
then you can use useNavigate() hooks on hostApp to update path. i hope below code helps your issue

subApp
`router.subscribe((state) => {
    onNavigate(state);
  });`

hostApp
`mount({
     ...,
      onNavigate: (state: RouteState) => {
        navigate(state.location.pathname);
      },
    });`

also this stackoverflow link may be help; https://stackoverflow.com/questions/74165853/react-router-dom-v6-4-doesnt-allow-history-listen-prior-suggestions-deprecated

**Comment by ahmetartana** [comment]
Thank you for your support @yasiinakbulut58
